From 76e402e8b446861adf45f8ece54d616808586b71 Mon Sep 17 00:00:00 2001
From: Alexandre Derumier <aderumier@odiso.com>
Date: Wed, 16 May 2018 02:11:45 +0200
Subject: [PATCH 01/12] don't remove (tap|veth|fwpr) interfaces from bridge on
 ifup bridge

as proxmox don't defined them in /etc/network/interfaces

Signed-off-by: Thomas Lamprecht <t.lamprecht@proxmox.com>
Signed-off-by: Alexandre Derumier <aderumier@odiso.com>
---
 ifupdown2/addons/bridge.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ifupdown2/addons/bridge.py b/ifupdown2/addons/bridge.py
index 954e462..ff98b18 100644
--- a/ifupdown2/addons/bridge.py
+++ b/ifupdown2/addons/bridge.py
@@ -443,6 +443,7 @@ class bridge(Bridge, moduleBase):
             "bridge-ports-condone-regex": {
                     "help": "bridge ports to ignore/condone when reloading config / removing interfaces",
                     "required": False,
+                    "default": "^(tap|veth|fwpr)",
                     "example": ["bridge-ports-condone-regex ^[a-zA-Z0-9]+_v[0-9]{1,4}$"]
             },
             "bridge-vlan-vni-map": {
@@ -1152,6 +1153,9 @@ class bridge(Bridge, moduleBase):
         # up a compiled regex to be used in a match later. This way we try to avoid
         # a race condition where an (possibly VM) interface is created after this
         # function has been called but before the bridgeports are validated.
+        if not bridge_port_condone_regex:
+            bridge_port_condone_regex = self.get_attr_default_value('bridge-ports-condone-regex')
+
         if bridge_port_condone_regex:
             if get_string:
                 return bridge_port_condone_regex
-- 
2.30.2

